### PR TITLE
Create README.md

### PR DESCRIPTION
**Pull request by barm**
Add basic readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+We use submodules to manage components. To download the deps, follow the instructions below:
+
+```
+git submodules init
+git submodules update
+```
+
+## Environment setup
+
+Follow [ESP32 installation guide](https://docs.espressif.com/projects/esp-idf/en/v4.3.1/esp32/get-started/index.html#installation-step-by-step) to setup your local environment.
```
Comment (zeroomega) [inline README.md]: The project now requires v5 or above. So we should update this link to ESP-IDF 5

File: README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Enable BT 
+
+```
+idf.py menuconfig
+```
+
+In order for this to launch, you may need to follow the on-screen instructions to install `windows-curses`. 
+
+In menufconfig, select `Component config -> Bluetooth` and enable Bluetooth (Bluedroid). Save and exit.
```
Comment (zeroomega) [inline README.md]: I believe the missing feature causing your build error is that BLE 4.2 feature is not enabled. It is under menuconfig as well. May worth mentioning. In fact with latest `sdkconfig.default` patch, these should automatically enabled now.

Comment (barm) [reply README.md]: Now that this is included in `sdkconfig.defaults`, I think we just need to instruct people to run `idf.py menuconfig` and immediately exit right?

Comment (zeroomega) [reply README.md]: I think so.
I suspect even idf.py menuconfig step can be omit, but I don't have a machine to test it from start to finish now. There is no harm to run menuconfig anyway so it is OK we keep it.

Comment (barm) [reply README.md]: Confirmed it works without explicitly running menuconfig.

**Pull request merged**